TYPO3.Flow/Classes/TYPO3/Flow/Http/Component/TrustedProxiesComponent.php

```diff
@@ -50,14 +50,21 @@ public function handle(ComponentContext $componentContext)
         }
 
         $hostHeader = $this->getFirstTrustedProxyHeaderValue(self::HEADER_HOST, $trustedRequest);
+        $portFromHost = null;
         if ($hostHeader !== null) {
+            $portSeparator = strrpos($hostHeader, ':');
+            if ($portSeparator > 0) {
```

Comment:
I doubt that we support IPv6 at all, as we'd have to strip of the square brackets around these to get the port (see https://stackoverflow.com/questions/186829/how-do-ports-work-with-ipv6).

Anyways, good call, so this should work even for `:<just-the-port>`:
```php
$portSeparatorIndex = strrpos($hostHeader, ':');
if ($portSeparatorIndex !== false) {
    $portFromHost = substr($hostHeader, $portSeparatorIndex + 1);
    $trustedRequest->getUri()->setPort($portFromHost);
    $hostHeader = substr($hostHeader, 0, $portSeparatorIndex);
}
$trustedRequest->getUri()->setHost($hostHeader);
```
IMO https://github.com/neos/flow-development-collection/pull/1056#pullrequestreview-56697249 is more important though